kdtree.py

```diff
@@ -219,7 +219,8 @@ def __init__(self, data=None, left=None, right=None, axis=None,
         sel_axis(axis) is used when creating subnodes of the current node. It
         receives the axis of the parent node and returns the axis of the child
         node. """
-
+        if type(data) == list:
```

Comment:
Is there any specific data why you're checking for type ``list``? Could you please also add a comment to describe the intention?